src/internal/connector/exchange/exchange_data_collection.go

```diff
@@ -139,33 +144,83 @@ func (col *Collection) populateByOptionIdentifier(
 		return
 	}
 
+	limitCh := make(chan struct{}, urlPrefetchChannelBufferSize)
+	defer close(limitCh)
+
+	type uerr struct {
+		user string
+		err  error
+	}
+
+	errCh := make(chan uerr, len(col.jobs))
+	defer close(errCh)
+
+	ffail := int64(0)
 	for _, identifier := range col.jobs {
-		response, err := query(ctx, col.service, user, identifier)
-		if err != nil {
-			errs = support.WrapAndAppendf(user, err, errs)
+		if ffail > 0 {
+			break
+		}
+		limitCh <- struct{}{}
+
+		wg.Add(1)
+
+		go func(identifier string) {
+			defer func() { <-limitCh }()
+			defer wg.Done()
```

Comment:
These can be grouped. 

Additionally, a comment would be helpful.

Reply:
Any specific reason for grouping them? Let me add comments to it though.

Reply:
I've added comments to the channel creation.